src/components/MultiGestureCanvas/index.js

```diff
@@ -262,7 +274,7 @@ function ImageTransformer({imageWidth, imageHeight, imageScaleX, imageScaleY, sc
             zoomScale.value = withSpring(doubleTapScale, SPRING_CONFIG);
             pinchScaleOffset.value = doubleTapScale;
         },
-        [scaledImageWidth, scaledImageHeight, canvasWidth, canvasHeight],
+        [contentScaling.scaledWidth, contentScaling.scaledHeight, canvasSize.width, canvasSize.height],
```

Comment:
Oh yea, the reason for that is that the "double tap zoom scale" is capped by either the `maxZoomScale` from the provided range or the scale that's needed for fitting inside the canvas. This is done by something like this: `Math.max(zoomRange.max, fittingZoomScale)`.

I think this is more of a design question in general, if double tap should always fit the content, especially with very wide/tall images...

Reply:
> Can we include `doubleTapScale` as a dependency? Sometimes, when I double-tap, it only zooms in to 3x instead of filling the canvas.

Double tap now always "zooms to fill"